scripts/aws/monitor.py

```diff
@@ -0,0 +1,187 @@
+    parser.add_argument("--splunk-host", required=True)
+    parser.add_argument("--splunk-token", required=True)
+
+    args = parser.parse_args()
+    session = botocore.session.Session(profile=args.aws_profile)
+
+    sns_client = session.create_client('sns', args.aws_region)
+    cw = session.create_client('cloudwatch', args.aws_region)
+
+    create_api_alarm(cw, 'api_count', 'Count', 'ApiGateway',
+                     'Average', 'GreaterThanOrEqualToThreshold',
+                     'Average API count for a period of 5 min', 50, 300, 1,
+                     [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_latency', 'Latency', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average API Latency for a period of 5 min',
+                     3, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_errors_4xx', '4XXError', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average 4XX errors for a period of 5 min',
+                     4, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    create_api_alarm(cw, 'api_errors_5xx', '5XXError', 'ApiGateway', 'Average',
+                     'GreaterThanOrEqualToThreshold', 'Average 5XX errors for a period of 5 min',
+                     4, 300, 1, [{'Name': 'ApiName', 'Value': args.api_gw}, {'Name': 'Label', 'Value': args.api_gw}],
+                     get_topic_arn(sns_client, 'aws-non-critical-alert'))
+
+    iam_client = session.create_client('iam', args.aws_region)
+    role_arn = create_role(iam_client, 'lambda_basic_execution_monitor',
+                           '{"Version": "2012-10-17","Statement": [{"Effect": "Allow","Principal": '
+                           '{"Service": "lambda.amazonaws.com"},"Action": "sts:AssumeRole"}]}',
+                           open(os.path.join(os.path.dirname(__file__), 'lambda_exec_policy.json')).read())
+
+    logging.info('Waiting for the newly created role to be available')
+    # Sleep for 10 seconds to allow the role created above to be avialable for lambda function creation
+    time.sleep(10)
+    lambda_client = session.create_client('lambda', args.aws_region)
+    tmpdirname = tempfile.mkdtemp()
+    zip_file_name = create_lambda_function_zip(tmpdirname, args.splunk_host, args.splunk_token)
+    create_lambda_function(lambda_client, 'cloudwatch-logs-splunk', 'nodejs4.3', role_arn,
+                           'index.handler', zip_file_name,
+                           'Demonstrates logging events to Splunk HTTP Event '
+                           'Collector.', 10, 512)
```

Comment:
What variables are 10 and 512 and why are they fixed values instead of configurable tuning parameters? (Based on my knowledge of Lambda I assume that this is "max lambda execution time" and "max lambda memory utilization".)